lib/matplotlib/figure.py

```diff
@@ -1528,635 +965,1865 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
+
+    Typically instantiated using `.Figure.add_subfigure` or
+    `.SubFigure.add_subfigure`, or `.SubFigure.subfigures`.  A subfigure has
+    the same methods as a figure except for those particularly tied to the size
+    or dpi of the figure, and is confined to a prescribed region of the figure.
+    For example the following puts two subfigures side-by-side::
+
+        fig = plt.figure()
+        sfigs = fig.subfigures(1, 2)
+        axsL = sfigs[0].subplots(1, 2)
+        axsR = sfigs[1].subplots(2, 1)
+
+    See :doc:`/gallery/subplots_axes_and_figures/subfigures`
+    """
+
+    def __init__(self, parent, subplotspec, *,
+                 facecolor=None,
+                 edgecolor=None,
+                 linewidth=0.0,
+                 frameon=None):
+        """
+        Parameters
+        ----------
+        parent : `.figure.Figure` or `.figure.SubFigure`
+            Figure or subfigure that contains the SubFigure.  SubFigures
+            can be nested.
+
+        subplotspec : `.gridspec.SubplotSpec`
+            Defines the region in a parent gridspec where the subfigure will
+            be placed.
+
+        facecolor : default: :rc:`figure.facecolor`
+            The figure patch face color.
+
+        edgecolor : default: :rc:`figure.edgecolor`
+            The figure patch edge color.
+
+        linewidth : float
+            The linewidth of the frame (i.e. the edge linewidth of the figure
+            patch).
+
+        frameon : bool, default: :rc:`figure.frameon`
+            If ``False``, suppress drawing the figure background patch.
+        """
+        super().__init__()
+        if facecolor is None:
+            facecolor = mpl.rcParams['figure.facecolor']
+        if edgecolor is None:
+            edgecolor = mpl.rcParams['figure.edgecolor']
+        if frameon is None:
+            frameon = mpl.rcParams['figure.frameon']
+
+        self._subplotspec = subplotspec
+        self._parent = parent
+        self.figure = parent.figure
+        # subfigures use the parent axstack
+        self._axstack = parent._axstack
+        self.subplotpars = parent.subplotpars
+        self.dpi_scale_trans = parent.dpi_scale_trans
+        self._axobservers = parent._axobservers
+        self.dpi = parent.dpi
+        self.canvas = parent.canvas
+        self.transFigure = parent.transFigure
+        self.bbox_relative = None
+        self._redo_transform_rel_fig()
+        self.figbbox = self._parent.figbbox
+        self.bbox = TransformedBbox(self.bbox_relative,
+                                    self._parent.transSubfigure)
+        self.transSubfigure = BboxTransformTo(self.bbox)
+
+        self.patch = Rectangle(
+            xy=(0, 0), width=1, height=1, visible=frameon,
+            facecolor=facecolor, edgecolor=edgecolor, linewidth=linewidth,
+            # Don't let the figure patch influence bbox calculation.
+            in_layout=False, transform=self.transSubfigure)
```

Comment:
In `Figure.__init__`, it calls `self._set_artist_props(self.patch)`, which would set the transform, among other things. Should that not happen here?